### ISSUE TITLE
step-definitions.ts assumes that code outside the project root can't be valid

### ISSUE THREAD

**Issue by cellog** [badeball/cypress-cucumber-preprocessor#719]
### Current behavior

We have a project which defines an integration folder outside the project root (it is a sibling directory) as `../src`. This works great. However, we've been stuck on the older browserify with the original `cypress-cucumber-preprocessor`. For more explanation: we are co-locating tests that use storybook as the source with the component they are testing. Because we have both e2e and integration tests, we use the `--project` flag to set the root directory to the location of the specific test boilerplate (either integration or e2e).

I just got things to work with `@badeball/cypress-cucumber-preprocessor` (yay!) but the check that a step file is within the project root is failing. I don't think this is necessary at all, and can be safely removed.

### Desired behavior

The only check is whether the step is within the integration folder, just like Cypress does.

### Test code to reproduce

Here's the patch that `patch-package` made. I can make a PR against the source if you'd like.

```diff
diff --git a/node_modules/@badeball/cypress-cucumber-preprocessor/lib/step-definitions.js b/node_modules/@badeball/cypress-cucumber-preprocessor/lib/step-definitions.js
index 808b126..ef12071 100644
--- a/node_modules/@badeball/cypress-cucumber-preprocessor/lib/step-definitions.js
+++ b/node_modules/@badeball/cypress-cucumber-preprocessor/lib/step-definitions.js
@@ -40,9 +40,6 @@ function getStepDefinitionPatterns(configuration, filepath) {
     if (!(0, is_path_inside_1.default)(filepath, fullIntegrationFolder)) {
         throw new Error(`${filepath} is not inside ${fullIntegrationFolder}`);
     }
-    if (!(0, is_path_inside_1.default)(filepath, configuration.cypress.projectRoot)) {
-        throw new Error(`${filepath} is not inside ${configuration.cypress.projectRoot}`);
-    }
     (0, debug_1.default)(`looking for step definitions using ${util_1.default.inspect(configuration.preprocessor.stepDefinitions)}`);
     const filepathReplacement = trimFeatureExtension(path_1.default.relative(fullIntegrationFolder, filepath));
     (0, debug_1.default)(`replacing [filepath] with ${util_1.default.inspect(filepathReplacement)}`);
```

### Versions

* **Cypress version**: doesn't matter but we are on 9.6.0. This has worked since version 6 I think
* **Preprocessor version**: 9.2.1
* **Node version**: 14.15.4

**Comment by badeball** [comment]
You're right, this _does_ seem unnecessary. I think it originates as a sanity check for when I initially wrote the code. However, I see no reason for why steps should be required to be within the project root.

**Comment by badeball** [comment]
I've released v10.0.2 which removes the mentioned fix. Hope that helps.

**Comment by cellog** [comment]
NICE, thanks! That's one `patch-package` I can remove 🤣
